storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -944,13 +945,24 @@ bool Redo_Log_Data_Manager::start() {
   last_checkpoint_lsn = start_checkpoint_lsn;
   stop_lsn = 0;
 
+  if (opt_lock_ddl_per_table) {
+    mdl_lock_tables();
+  }
+
   bool finished = false;
   while (!finished) {
     if (!copy_once(false, &finished)) {
       return (false);
     }
   }
 
+  /*
+   * From this point forward, recv_parse_or_apply_log_rec_body should fail if
+   * MLOG_INDEX_LOAD event is parsed as its not safe to continue the backup
+   * in any situation (with or without --lock-ddl-per-table).
+   */
+  mdl_taken = true;
```

Comment:
One doubt, if opt_lock_ddl_per_table is OFF (mdl is not taken), we still set mdl_taken= true. Is ithis intentional?

Reply:
Ok, with LOCK INSTANCE/LOCK_TABLES_FOR_BACKUP taken earlier, we wouldn't see MLOG_INDEX_LOAD. So the setting mdl_taken=true is OK for these scenarios.